tests/integration/test_schema.py

```diff
@@ -990,23 +1102,29 @@ async def test_schema_types(registry_async_client, trail):
 
 @pytest.mark.parametrize("trail", ["", "/"])
 async def test_schema(registry_async_client, trail):
-    subject = new_random_name("subject")
-    schema_str = '{"type": "string"}'
+    subject_name_factory = create_id_factory(f"test_schema-{trail}")
+    unique_field_factory = create_id_factory(trail)
+
+    subject = subject_name_factory()
+    schema_str = '{"type": "string", "unique" : "%s"}' % unique_field_factory()
     res = await registry_async_client.post(
         f"subjects/{subject}/versions{trail}",
         json={"schema": schema_str},
     )
     assert res.status == 200
     assert "id" in res.json()
     schema_id = res.json()["id"]
+
     res = await registry_async_client.get(f"schemas/ids/{schema_id}{trail}")
     assert res.status_code == 200
     assert res.json()["schema"] == schema_str
 
+    await assert_schema_versions(registry_async_client, trail, schema_id, [(subject, 1)])
+
     # repost same schema again to see that a new id is not generated but an old one is given back
     res = await registry_async_client.post(
         f"subjects/{subject}/versions{trail}",
-        json={"schema": '{"type": "string"}'},
+        json={"schema": '{"type": "string", "unique" : "%s"}' % unique_field_factory()},
```

Comment:
This change is breaking the test, this schemas is not supposed to be different, as per the comment above.

Reply:
Yes, a bug in the code. Fixing.